server/routes/barcode/PdfController.ts

```diff
@@ -36,4 +40,34 @@ export default class PdfController {
     const view = new PdfControllerView(req.session.pdfForm, req.flash('errors'))
     return res.render('pages/barcode/pdf/print-coversheets', { ...view.renderArgs })
   }
+
+  async submitPrintCoverSheet(req: Request, res: Response): Promise<void> {
+    if (!req.session.recipients) {
+      return res.redirect('/barcode/find-recipient')
+    }
+    if (!req.session.pdfForm) {
+      return res.redirect('/barcode/pdf/select-envelope-size')
+    }
+
+    const barcodeImages = await Promise.all(
```

Comment:
This is the loop we will need in `GenerateBarcodeController` when we officially support multiple recipients
We basically need to loop through all recipients, calling `createBarcodeService.generateBarcode` on each one. The minor complexity is that `createBarcodeService.generateBarcode` returns a `Promise`, so we need to use `Promise.all` to wait for them all to be done.

And then there is the question of what happens if one of them fails .... should it fail them all, or just the one? ... One for another day ...

Reply:
Hmmm one fails all fail doesn't sound great!

Reply:
Indeed. At the moment I think my catch block will essentially skip the one that fails and will generate a PDF with all the others in it .... but we need to consider how we handle that to the user .... how do we tell them that their PDF contains n-1 barcodes? how do we tell them which one failed? how do we tell them that they all failed? (ie. they could all fail if there are multiple recipients, or it could just be the one recipient that fails)

In fact I'll add 'error handling' to my list of questions ....